python/equistore-operations/equistore/operations/_utils.py

```diff
@@ -11,7 +11,7 @@ class NotEqualError(Exception):
     pass
 
 
-def _check_same_keys(a: TensorMap, b: TensorMap, fname: str):
+def _check_same_keys(a: TensorMap, b: TensorMap, fname: str) -> str:
```

Comment:
these functions are used elsewhere in code that assumes they raise. So there should also be a `_raise` and `_impl` version for these functions.